libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/api/MapboxRouteLineApi.kt

```diff
@@ -739,12 +921,34 @@ class MapboxRouteLineApi(
                 trafficLineExpressionProducer,
                 routeLineExpression,
                 routeLineCasingExpression,
-                alternativeRoute1TrafficExpressionProducer,
-                alternativeRoute2TrafficExpressionProducer,
+                alternateRoute1TrafficExpressionProducer,
+                alternateRoute2TrafficExpressionProducer,
                 alternativeRoute1FeatureCollection,
                 alternativeRoute2FeatureCollection,
                 wayPointsFeatureCollection
             )
         )
     }
+
+    internal val alternativelyStyleSegmentsNotInLeg: (
+        activeLegIndex: Int,
+        segments: List<RouteLineExpressionData>
+    ) -> List<RouteLineExpressionData> =
+        { activeLegIndex: Int, segments: List<RouteLineExpressionData> ->
+            segments.parallelMap(
```

Comment:
I didn't benchmark it but there are typically a lot of items in the collection (especially for long routes) and they can be operated on independently. This method gets called during a route progress update so that it can be ready when a user transitions from one leg to another. I thought it was worth operating on them in parallel.